File: lib/models/Shift.dart
class Shift {
  final int shiftId;
  final String name;
  final String from;
  final String to;

  Shift({required this.shiftId, required this.name, required this.from, required this.to});

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      shiftId: json['shift_id'],
      name: json['name'],
      from: json['from'],
      to: json['to'],
    );
  }
}

File: service/ShiftService/models/Shift.js
const mongoose = require('mongoose');

const ShiftSchema = new mongoose.Schema({
    shift_id: { type: Number, unique: true }, 
    name: { type: String, required: true },
    from: { type: String, required: true },
    to: { type: String, required: true },
    time: { type: Date, default: Date.now },
    secretCode: { type: String, required: true },
}, { collection: 'shifts' });

module.exports = mongoose.model('Shift', ShiftSchema);
